[css-grid] adjustGridPositionsFromStyle should set a subgrid's indefinite span to use the null string

sammygill opened the pull request:
#### c2e83fc185932b1b09c3ef4d75468cf47b2515d2
<pre>
[css-grid] adjustGridPositionsFromStyle should set a subgrid&apos;s indefinite span to use the null string
<a href="https://bugs.webkit.org/show_bug.cgi?id=260527">https://bugs.webkit.org/show_bug.cgi?id=260527</a>
rdar://114271485

Reviewed by Matt Woodrow.

If a subgrid has an indefinite span in the subgridded dimension,
adjustGridPositionsFromStyle will attempt to set the span size based
on the the number of named lines that it has. This is done with a call
similar to span.setSpanPosition(std::max(1, lineCount - 1), emtpyString()).

This code presumably does not want to set the m_namedGridLine portion
of the GridPosition as we are not actually referencing a named grid
line. However, we should probably be using String() here instead of
emptyString() to set that value to the null string as the rest of the
code checks against !position.namedGridLine().isNull() when attempting
to use the named grid line. This change helps us avoid entering code
guarded by that type of logic when we probably do not want to.

* LayoutTests/TestExpectations:
* Source/WebCore/rendering/style/GridPositionsResolver.cpp:
(WebCore::adjustGridPositionsFromStyle):

Canonical link: <a href="https://commits.webkit.org/268510@main">https://commits.webkit.org/268510@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/cc51cef7f320498f591b0c20a27696259f652981

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19932 "Failed to checkout and rebase branch from PR 17941") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20357 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20978 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21827 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/18616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/20168 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23614 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20508 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/20154 "Failed to checkout and rebase branch from PR 17941") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/20125 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/17341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22681 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/17295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/18136 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24401 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/18372 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/18312 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/22388 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/18902 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/16029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18073 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/22422 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2437 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/18726 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/rendering/style/GridPositionsResolver.cpp:
```diff
@@ -381,10 +381,10 @@ static void adjustGridPositionsFromStyle(const RenderBox& gridItem, GridTrackSiz
         if (initialPosition.isAuto()) {
             // Set initial position to span <line names - 1>
-            initialPosition.setSpanPosition(std::max(1, lineCount - 1), emptyString());
+            initialPosition.setSpanPosition(std::max(1, lineCount - 1), String());
```
Changing to the String() looks fine. Was not super clear to me why this was set to an empty string in the first place from reading the past commits.


The grid spec was not super helpful either from a read through.

We could also do checks for `isEmpty` instead of `isNull`, but I am not sure we would gain much benefit.

sammygill committed:
[css-grid] adjustGridPositionsFromStyle should set a subgrid's indefinite span to use the null string
https://bugs.webkit.org/show_bug.cgi?id=260527
rdar://114271485

Reviewed by Matt Woodrow.

If a subgrid has an indefinite span in the subgridded dimension,
adjustGridPositionsFromStyle will attempt to set the span size based
on the the number of named lines that it has. This is done with a call
similar to span.setSpanPosition(std::max(1, lineCount - 1), emtpyString()).

This code presumably does not want to set the m_namedGridLine portion
of the GridPosition as we are not actually referencing a named grid
line. However, we should probably be using String() here instead of
emptyString() to set that value to the null string as the rest of the
code checks against !position.namedGridLine().isNull() when attempting
to use the named grid line. This change helps us avoid entering code
guarded by that type of logic when we probably do not want to.

* LayoutTests/TestExpectations:
* Source/WebCore/rendering/style/GridPositionsResolver.cpp:
(WebCore::adjustGridPositionsFromStyle):

Canonical link: https://commits.webkit.org/268510@main
